File: lines.py
import os


def count_lines_in_file(file_path):
    with open(file_path, 'r', encoding='utf-8') as file:
        num_lines = 0
        for line in file:
            line = line.strip()
            if len(line) > 0 and not line.startswith("//"):
                num_lines += 1
        return num_lines


def count_lines_in_directory(directory_path, file_extension):
    num_lines = 0
    for root, dirs, files in os.walk(directory_path):
        for file in files:
            if file.endswith(file_extension):
                file_path = os.path.join(root, file)
                num_lines += count_lines_in_file(file_path)
    return num_lines

total_lines = count_lines_in_directory(".", ".go")
print(f"Total number of lines in *.go files: {total_lines}")

File: lines.go
package main

import (
	"bufio"
	"os"
	"path/filepath"
	"strings"
)

func countLinesInFile(filePath string) (int, error) {
	file, err := os.Open(filePath)
	if err != nil {
		return 0, err
	}
	defer func() { _ = file.Close() }()

	var numLines int
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := strings.TrimSpace(scanner.Text())
		if len(line) > 0 && !strings.HasPrefix(line, "//") {
			numLines++
		}
	}
	if err = scanner.Err(); err != nil {
		return 0, err
	}

	return numLines, nil
}

func countLinesInDirectory(directoryPath, fileExtension string) (int, error) {
	var numLines int
	err := filepath.Walk(directoryPath, func(path string, info os.FileInfo, err error) error {
		if err != nil {
			return err
		}
		if !info.IsDir() && strings.HasSuffix(info.Name(), fileExtension) {
			linesInFile, err := countLinesInFile(path)
			if err != nil {
				return err
			}
			numLines += linesInFile
		}
		return nil
	})
	if err != nil {
		return 0, err
	}

	return numLines, nil
}

//func main() {
//	totalLines, err := countLinesInDirectory(".", ".go")
//	if err != nil {
//		fmt.Println("Error:", err)
//		return
//	}
//	fmt.Printf("Total number of lines in *.go files: %d\n", totalLines)
//}
